cli/cliui/cliui.go

```diff
@@ -117,48 +101,150 @@ func ifTerm(fmt pretty.Formatter) pretty.Formatter {
+		// are deterministic.
+		color = termenv.Ascii
+	}
+
 	// We do not adapt the color based on whether the terminal is light or dark.
 	// Doing so would require a round-trip between the program and the terminal
 	// due to the OSC query and response.
-	DefaultStyles = Styles{
-		Code: pretty.Style{
-			ifTerm(pretty.XPad(1, 1)),
-			pretty.FgColor(Red),
-			pretty.BgColor(color.Color("#2c2c2c")),
-		},
-		DateTimeStamp: pretty.Style{
-			pretty.FgColor(color.Color("#7571F9")),
-		},
-		Error: pretty.Style{
-			pretty.FgColor(Red),
-		},
-		Field: pretty.Style{
-			pretty.XPad(1, 1),
-			pretty.FgColor(color.Color("#FFFFFF")),
-			pretty.BgColor(color.Color("#2b2a2a")),
-		},
-		Keyword: pretty.Style{
-			pretty.FgColor(Green),
-		},
-		Placeholder: pretty.Style{
-			pretty.FgColor(color.Color("#4d46b3")),
-		},
-		Prompt: pretty.Style{
-			pretty.FgColor(color.Color("#5C5C5C")),
-			pretty.Wrap("> ", ""),
-		},
-		Warn: pretty.Style{
-			pretty.FgColor(Yellow),
+	DefaultStyles = Styles{Wrap: pretty.Style{pretty.LineWrap(80)}}
+
+	defaultTheme := userThemeConfig{
+		Colors: map[string]string{
+			"green":  "#04B575",
+			"red":    "#ED567A",
+			"yellow": "#ECFD65",
+			"blue":   "#5000ff",
 		},
-		Wrap: pretty.Style{
-			pretty.LineWrap(80),
+		Styles: &userThemeStyles{
+			Code: &userThemeStyle{
+				FgColor: "$red",
+				BgColor: "#2C2C2C",
+			},
+			DateTimeStamp: &userThemeStyle{
+				FgColor: "#7571F9",
+			},
+			Error: &userThemeStyle{
+				FgColor: "$red",
+			},
+			Field: &userThemeStyle{
+				FgColor: "#FFFFFF",
+				BgColor: "#2B2A2A",
+			},
+			FocusedPrompt: &userThemeStyle{
+				FgColor: "$blue",
+			},
+			Keyword: &userThemeStyle{
+				FgColor: "$green",
+			},
+			Placeholder: &userThemeStyle{
+				FgColor: "#4d46b3",
+			},
+			Prompt: &userThemeStyle{
+				FgColor: "#5C5C5C",
+			},
+			Warn: &userThemeStyle{
+				FgColor: "$yellow",
+			},
 		},
 	}
 
-	DefaultStyles.FocusedPrompt = append(
-		DefaultStyles.Prompt,
-		pretty.FgColor(Blue),
-	)
+	configDir := config.DefaultDir()
+	if dir := os.Getenv("CODER_CONFIG_DIR"); dir != "" {
+		configDir = dir
+	}
+
+	if theme, err := os.ReadFile(path.Join(configDir, "theme.toml")); err == nil {
+		_ = toml.Unmarshal(theme, &defaultTheme)
+	}
+
+	applyUserTheme(defaultTheme)
+}
+
+type userThemeStyle struct {
+	BgColor string `toml:"background"`
+	FgColor string `toml:"foreground"`
+}
+
+type userThemeStyles struct {
+	Code          *userThemeStyle `toml:"code"`
+	DateTimeStamp *userThemeStyle `toml:"datetimestamp"`
+	Error         *userThemeStyle `toml:"error"`
+	Field         *userThemeStyle `toml:"field"`
+	FocusedPrompt *userThemeStyle `toml:"focusedprompt"`
+	Keyword       *userThemeStyle `toml:"keyword"`
+	Placeholder   *userThemeStyle `toml:"placeholder"`
+	Prompt        *userThemeStyle `toml:"prompt"`
+	Warn          *userThemeStyle `toml:"warn"`
+}
+
+type userThemeConfig struct {
+	Colors map[string]string `toml:"colors"`
+	Styles *userThemeStyles  `toml:"styles"`
```

Comment:
I wonder if it would be simpler to just omit "colors" and use styles/color codes only?

In a way it's cool to be able to change e.g. red and have multiple properties change, but since not all colors are addressable this way, I'm not convinced of its usefulness. Especially in cases where we also have a background color that's static.

Reply:
It would definitely be simpler. I'm happy to remove it if desired. My thinking was that since the previous styling had set colors and referenced those it would replicate that behaviour.

Reply:
Yeah, this caught me out initially until I realised what `$color` actually meant.

Reply:
Happy to back this part out then. Would we want to drop `Styles` and put all the configuration in this top `userThemeConfig` type?

Reply:
I've backed this out